### ISSUE TITLE
Version 4 hierarchical hostnames iterate in the opposite direction to paths

### ISSUE THREAD

**Issue by shabbyrobe** [thephpleague/url#91]
We read domain names from left to right and file paths from left to right, but hierarchically, domain names are right to left. The taxonomy of the host class suggests it is hierarchical (it is an `AbstractHierarchicalComponent`), in which case I would expect it to iterate right to left, instead of left to right as it currently does:

```
$url = \League\Uri\Schemes\Http::createFromString("http://three.two.one/one/two/three");
foreach ($url->host as $part) {
    echo $part."\n";
}

foreach ($url->path as $part) {
    echo $part."\n";
}
```

Output:
```
three
two
one
one
two
three
```

**Comment by nyamsprod** [comment]
Hi @shabbyrobe thanks for the feedback... the package will be updated to take into account your remark. That being said, because v4 was a large departure from v3. We have decided to create a new repo https://github.com/thephpleague/uri where the package is now in development. So the bug fix will be made there. 
TL;DR: the new package wil be release as League\Uri and League\Url will be deprecated when League\Uri is released. The fix will be done in League\uri

**Comment by nyamsprod** [comment]
The enhancement or bug fix has landed on the master branch of the https://github.com/thephpleague/uri repository
